actors/market/src/lib.rs

```diff
@@ -851,23 +828,24 @@ impl Actor {
                         state.last_updated_epoch = curr_epoch;
                         st.put_deal_states(rt.store(), &[(deal_id, state)])?;
 
-                        if let Some(ev) = updates_needed.get_mut(&next_epoch) {
-                            ev.push(deal_id);
-                        } else {
-                            updates_needed.insert(next_epoch, vec![deal_id]);
-                        }
+                        // Compute and record the next epoch in which this deal will be updated.
+                        // This epoch is independent of the deal's stated start and end epochs
+                        // in order to prevent intentional scheduling of many deals for the same
+                        // update epoch.
+                        let next_epoch = next_update_epoch(
+                            deal_id,
+                            rt.policy().deal_updates_interval,
+                            curr_epoch + 1,
+                        );
+                        new_updates_scheduled.entry(next_epoch).or_default().push(deal_id);
```

Comment:
It would be good to have a test covering scheduling multiple updates to the same next epoch.  Im guessing we won't get this be existing tests because we would need either too many deals or modify state.  The network will only hit this case during the first day of processing spreading out deal ops over the longer update interval.

Reply:
Added